File: daily_python/3191_Minimum_Operations_to_Make_Binary_Array_ElementsEqual_to_One_I.py
import os
import sys
from typing import List

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        for i in range(n - 2):
            if nums[i] == 0:
                ans += 1
                for d in range(i, i + 3):
                    nums[d] ^= 1
        return ans if (nums[n-2] == 1 and nums[n-1] == 1) else -1

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.minOperations(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/3191_Minimum_Operations_to_Make_Binary_Array_ElementsEqual_to_One_I.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;
        for (int i = 0; i < n - 2; i++){
            if (nums[i] == 0){
                ans++;
                for (int d = 0; d < 3; d++) nums[i+d] ^= 1;
            }
        }
        if (nums[n-2] == 1 && nums[n-1] == 1) return ans;
        else return -1;
    }
};

int main(){
    vector<int> nums = {0,1,1,1,0,0};
    // vector<int> nums = {0,1,1,1};
    Solution S;
    cout << S.minOperations(nums) << endl;
    return 0;
}